### ISSUE TITLE
tokenautocomplete crashes

### ISSUE THREAD

**Issue by Valodim** [open-keychain/open-keychain#1229]
```
 13381         AndroidRuntime  E  FATAL EXCEPTION: main
 13381         AndroidRuntime  E  Process: org.sufficientlysecure.keychain, PID: 13381
 13381         AndroidRuntime  E  java.lang.IllegalStateException: attempt to re-open an already-closed object: SQLiteQuery: SELECT keys.oid AS _id, keys.master_key_id, user_packets.user_id, keys.is_revoked, (keys.expiry IS NOT NULL AND keys.expiry < 1430156267) AS is_expired, verified, (EXISTS (SELECT * FROM keyrings_secret WHERE keyrings_secret.master_key_id = keys.master_key_id)) AS has_any_secret, (SELECT COUNT (*) FROM user_packets AS dups WHERE dups.master_key_id != keys.master_key_id AND dups.rank = 0 AND dups.user_id = user_packets.user_id) AS has_duplicate_user_id, kE.key_id AS has_encrypt, keys.fingerprint, keys.creation FROM keys INNER JOIN user_packets ON (keys.master_key_id = user_packets.master_key_id AND user_packets.rank = 0) LEFT JOIN certs ON (keys.master_key_id = certs.master_key_id AND certs.verified = 1) LEFT JOIN keys AS kE ON (kE.master_key_id = keys.master_key_id AND kE.is_revoked = 0 AND kE.can_encrypt = 1 AND ( kE.expiry IS NULL OR kE.expiry >= 1430156267 )) WHERE (keys.rank = 0) AND (has_encrypt NOT NULL AND is_expired = 0 AND keys.is_revoked = 0 AND user_id LIKE ?) GROUP BY keys.master_key_id ORDER BY user_packets.user_id ASC
 13381         AndroidRuntime  E  at android.database.sqlite.SQLiteClosable.acquireReference(SQLiteClosable.java:55)
 13381         AndroidRuntime  E  at android.database.sqlite.SQLiteQuery.fillWindow(SQLiteQuery.java:58)
 13381         AndroidRuntime  E  at android.database.sqlite.SQLiteCursor.fillWindow(SQLiteCursor.java:152)
 13381         AndroidRuntime  E  at android.database.sqlite.SQLiteCursor.onMove(SQLiteCursor.java:124)
 13381         AndroidRuntime  E  at android.database.AbstractCursor.moveToPosition(AbstractCursor.java:214)
 13381         AndroidRuntime  E  at android.database.CursorWrapper.moveToPosition(CursorWrapper.java:162)
 13381         AndroidRuntime  E  at android.support.v4.widget.CursorAdapter.getView(CursorAdapter.java:247)
 13381         AndroidRuntime  E  at android.widget.AbsListView.obtainView(AbsListView.java:2347)
 13381         AndroidRuntime  E  at android.widget.ListPopupWindow$DropDownListView.obtainView(ListPopupWindow.java:1706)
 13381         AndroidRuntime  E  at android.widget.ListView.measureHeightOfChildren(ListView.java:1270)
 13381         AndroidRuntime  E  at android.widget.ListPopupWindow.buildDropDown(ListPopupWindow.java:1181)
 13381         AndroidRuntime  E  at android.widget.ListPopupWindow.show(ListPopupWindow.java:568)
 13381         AndroidRuntime  E  at android.widget.AutoCompleteTextView.showDropDown(AutoCompleteTextView.java:1105)
 13381         AndroidRuntime  E  at android.widget.AutoCompleteTextView.updateDropDownForFilter(AutoCompleteTextView.java:974)
 13381         AndroidRuntime  E  at android.widget.AutoCompleteTextView.onFilterComplete(AutoCompleteTextView.java:956)
 13381         AndroidRuntime  E  at android.widget.Filter$ResultsHandler.handleMessage(Filter.java:285)
 13381         AndroidRuntime  E  at android.os.Handler.dispatchMessage(Handler.java:102)
 13381         AndroidRuntime  E  at android.os.Looper.loop(Looper.java:135)
 13381         AndroidRuntime  E  at android.app.ActivityThread.main(ActivityThread.java:5254)
 13381         AndroidRuntime  E  at java.lang.reflect.Method.invoke(Native Method)
 13381         AndroidRuntime  E  at java.lang.reflect.Method.invoke(Method.java:372)
 13381         AndroidRuntime  E  at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:903)
 13381         AndroidRuntime  E  at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:698)
```

it crashs regularly for me just typing random stuff in there.

**Comment by Valodim** [comment]
fixed in 91b774d22334b91c1867b867c515d1fcd31e34da I think

**Comment by Valodim** [comment]
fixed a litle more in e89dba8f245f59afd9e4e3da9c62f98befcb4d47
